ic-utils/src/interfaces/wallet.rs

```diff
@@ -163,22 +168,53 @@ impl<'agent> Canister<'agent, Wallet> {
 
     /// Forward a call to another canister, including an amount of cycles
     /// from the wallet.
-    pub fn call<'canister: 'agent, Out>(
+    pub fn call<'canister: 'agent, Out, M: ToString>(
```

Comment:
I'm curious: is this `M: ToString` here instead of `&str` because it's sometimes annoying to have to call `.to_string()/.as_str()/.whatever()`?

Reply:
Yeah this accepts any type which have the trait `ToString`, which includes `&str` and `String` (and maybe others?). So this is a shortcut to say "we accept both". I need to go through the whole repo and change our strings to that. I'm going to do a clean of these before version 0.2, which should also include an `IntoPrincipal` that can be either String, &str, or another Principal (or a `Canister<...>` for example).

Reply:
Also including things like a typed `RequestId` and better ways to wait on Requests (moving `call_and_wait()` into `call().and_wait()` (wait being on RequestId).

As I'm working between repos I see a lot more cracks like these that can improve the API greatly.